src/frontend/components/UI/AccessCodeInput/index.tsx

```diff
@@ -0,0 +1,75 @@
+import { TextInput, TextInputProps } from '@hyperplay/ui'
+import React, { useEffect, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import styles from './index.module.scss'
+
+export interface AccessCodeInputProps {
+  licenseConfigId?: number
+  accessCode: string
+  inputProps: TextInputProps
+  channelRequiresAccessCode?: boolean
+  setAccessCodeVerified: (val: boolean) => void
+}
+
+export function AccessCodeInput({
+  inputProps,
+  licenseConfigId,
+  accessCode,
+  channelRequiresAccessCode,
+  setAccessCodeVerified
+}: AccessCodeInputProps) {
+  const { t } = useTranslation()
+  const [errorText, setErrorText] = useState('')
+  const [successText, setSuccessText] = useState('')
+
+  useEffect(() => {
+    async function validateAccessCode() {
+      if (licenseConfigId === undefined) {
+        return
+      }
+
+      const result = await window.api.checkHyperPlayAccessCode(
+        licenseConfigId,
+        accessCode
+      )
+
+      setAccessCodeVerified(result.valid)
+
+      if (result.valid) {
+        setErrorText('')
+        setSuccessText(
+          t(
+            'hyperplay.accesscodes.success.validation',
+            'Success! Access code is valid'
+          )
+        )
+      } else {
+        setSuccessText('')
+        setErrorText(
+          t('hyperplay.accesscodes.error.validation', 'Access code is invalid')
+        )
+      }
+    }
+
+    if (accessCode && channelRequiresAccessCode) validateAccessCode()
+    else {
+      setErrorText('')
+      setSuccessText('')
+      setAccessCodeVerified(false)
+    }
+  }, [accessCode, licenseConfigId, channelRequiresAccessCode])
```

Comment:
you already know what I'm gonna say lol

we can a mutation for this

this PR is old though, we can keep this

Reply:
LOL

yeah, I promise that I will get back to it and use React Query, but since this is already in testing I will keep as it is for now.